src/std/fs.ab

```diff
@@ -76,3 +78,37 @@ pub fun change_owner(user: Text, path: Text): Bool {
 
     return false
 }
+
+/// Escapes all characters in the passed-in glob except "*", "?" and "/",
+/// to prevent injection attacks.
+fun escape_non_glob_chars(path: Text): Text {
+    return replace_regex(path, "\([^*?/]\)", "\\\\\1")
+}
+
+/// Finds all files matching the passed-in glob; uses `eval` because Bash
+/// does not expand environment variables in `for file in $path`. Also
+/// checks that files actually exist because Bash lists `*.xyz` as a file
+/// in `for file in *.xyz` if there are no matches.
+fun glob_and_ignore_missing(path: Text): [Text]? {
+    let files = $eval "for file in {path}; do [ -e \\\"\\\$file\\\" ] && echo \\\"\\\$file\\\"; done"$?
+    return split(files, "\n")
+}
```

Comment:
Sorry for going back and forth on this, but I don't like how the glob implementation function is separate from the escaping logic. This is a little bit dangerous as some future contributor might make use of this function without escaping the inputs. While this would likely be caught by reviewers, I don't like risking it.

Instead, lets merge all logic (incl. escaping) back into the `glob_multiple` function. Then lets have the regular `glob` function create an array with the single path it receives, and pass it to `glob_multiple`.

Inside of `glob_multiple` you can test if the length of the array is `1`, in which case you can skip joining its items.

Reply:
Heh. I'll change it back, but hopefully this will be the last one!

Reply:
Okay, refactored again; hopefully this will be the last time!  Note, I kept the private `escape_non_glob_chars()` function to avoid repetition, but that shouldn't cause any problems.